projects/core/koin-core/src/commonMain/kotlin/org/koin/core/module/Module.kt

```diff
@@ -235,12 +235,19 @@ operator fun List<Module>.plus(module: Module): List<Module> = this + listOf(mod
  * Run through the module list to flatten all modules & submodules
  */
 @OptIn(KoinInternalApi::class)
-fun flatten(modules: List<Module>): Set<Module> {
-    fun flat(modules: List<Module>, newModules: MutableSet<Module>){
-        modules.forEach{
-            newModules += it
-            flat(it.includedModules,newModules)
+fun flatten(modules: List<Module>): Set<Module> {    
```

Comment:
No action needed: that is nice and it is very similar to the original `flatten` function I wrote when introducing the `includes` feature [here](https://github.com/InsertKoinIO/koin/pull/1210/files#diff-e772f920a0a8f0cef44eb951fbb66027f8ee3b191a8cd91bc25b5f188ffaf837R338-R357) (for context).